tests/util.js

```diff
@@ -200,5 +203,81 @@ export default {
 				}
 			],
 		},
+		{
+			name: "canRegisterProperty()",
+
+			beforeEach () {
+				let iframe = document.createElement("iframe");
+				document.body.append(iframe);
+
+				this.data.iframe = iframe;
+			},
+
+			async run (source, ...properties) {
+				if (source === "inline style" || source === "adopted stylesheet") {
```

Comment:
This is pretty complicated, and I'm not sure the complexity is necessary. We don't need everything to be variable. Just import the static rules you've already defined and have each test include them in various ways. Conceptually, how the custom properties are registered is not really part of the test, it's part of the test setup.